### DISCUSSION TITLE
改变某个已知id组件的x-component-props的值

### DISCUSSION THREAD

**Discussion by 55555azhe** [alibaba/formily#1906] [Q&A 2.x]
formily外部怎么拿到内部组件的实例然后调用setComponentProps，或者怎么在外部动态改变某个已知id组件的x-component-props的值

**Comment by atzcl** [comment]
创建 [Form](https://core.formilyjs.org/zh-CN/api/entry/create-form) 实例进行操作

**Reply by 55555azhe** [reply]
文档都看了，没有对应的api，求具体的api是啥

**Reply by atzcl** [reply]
> 文档都看了，没有对应的api，求具体的api是啥

其实文档 demo 都有的，再认真看看，我这先给你写个简单的例子：[https://codesandbox.io/s/flamboyant-chebyshev-s4bwn?file=/App.tsx](https://codesandbox.io/s/flamboyant-chebyshev-s4bwn?file=/App.tsx)
